### PR TITLE
VAGOV-3157-SMTP-INTEGRATION

### PR DESCRIPTION
**Pull request by mbenziane**
https://github.com/department-of-veterans-affairs/va.gov-cms-devops/pull/75

### CODE REVIEW

**Comment by mbenziane** [comment]
rebuild

**Comment by mbenziane** [comment]
rebuild

**Comment by mbenziane** [comment]
rebuild

File: docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[PHP]
```
Comment (ElijahLynn) [inline docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl]: Can we pick a better name for this

File: docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+;;;;;;;;;;;;;;;
+; PHP Globals ;
```
Comment (ElijahLynn) [inline docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl]: Is all of this necessary for this PR?

File: Dockerfile-appserver [hunk 3]
```diff
@@ -33,3 +33,5 @@ COPY docker/templates/* /templates/
 COPY .composer-cache/README.md .composer-cache/composer-cache* /tmp/
 # Hack to only run composer install if no matching cache was downloaded.
 RUN if [ -f /tmp/composer-cache*.tar.gz ]; then cd /app && tar --extract --gzip --file /tmp/composer-cache*.tar.gz; else cd /app && composer install; fi
+# SMTP integration
```
Comment (ElijahLynn) [inline Dockerfile-appserver]: Can we move this before composer cache? Ideally we will be doing this in the base builds coming up.

Comment (mbenziane) [reply Dockerfile-appserver]: what is the difference?

File: docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+request_order = "GP"
+register_long_arrays = Off
+register_argc_argv = Off
+magic_quotes_gpc = Off
+enable_dl = Off
+allow_url_fopen = On
+realpath_cache_size = "800K"
+realpath_cache_ttl = "86400"
+disable_functions =
+sendmail_path=/usr/sbin/ssmtp -t
```
Comment (ElijahLynn) [inline docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl]: Can we just have this one sendmail line in its own ssmtp.ini file and copy that into conf.d?

File: Dockerfile-appserver (hunk 3 above)
Comment (ElijahLynn) [reply Dockerfile-appserver]: Suppose it doesn't matter right now, it is mainly for docker layer caching. Having the custom bits that change last is preferred for efficient docker layer cache.

File: docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl (hunk 1 above)
Comment (mbenziane) [reply docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl]: what name?

File: docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl (hunk 2 above)
Comment (mbenziane) [reply docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl]: exposing lando abstraction is a good idea for future customization

**Comment by mbenziane** [comment]
rebuild

File: docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl (hunk 2 above)
Comment (ElijahLynn) [reply docker/templates/zzz-lando-my-custom-ini-file-called-php.ini.tpl]: That is fine to abstract it but I don't think it should be lumped in with these settings. This is a good opportunity to break it out into its own ssmtp.ini file in conf.d.

**Comment by ElijahLynn** [comment]
build

**Comment by ElijahLynn** [comment]
build

**Comment by ElijahLynn** [comment]
build

**Comment by ElijahLynn** [comment]
build

**Pull request merged**